packages/jaeger-ui/src/components/SearchTracePage/SearchForm.js

```diff
@@ -264,16 +264,19 @@ export class SearchFormImpl extends React.PureComponent {
       services,
       submitting: disabled,
     } = this.props;
-    const selectedServicePayload = services.find(s => s.name === selectedService);
+    const servicesLoaded = services != null;
+    const servicesList = (servicesLoaded && services.map(v => ({ label: v.name, value: v.name }))) || [];
+    const selectedServicePayload = (servicesLoaded && services.find(s => s.name === selectedService)) || [];
     const opsForSvc = (selectedServicePayload && selectedServicePayload.operations) || [];
     const noSelectedService = selectedService === '-' || !selectedService;
     const tz = selectedLookback === 'custom' ? new Date().toTimeString().replace(/^.*?GMT/, 'UTC') : null;
+    const servicesCount = (servicesLoaded && services.length) || 'Loading...';
```

Comment:
this will result in `'Loading...'` if `services.length` is `0` regardless of if long polling services is enabled.